File: Algorithms/Python3.x/1365-How_Many_Numbers_Are_Smaller_Than_the_Current_Number.py
# Runtime: 56 ms, faster than 81.56% of Python3 online submissions for How Many Numbers Are Smaller Than the Current Number.
# Memory Usage: 13.9 MB, less than 100.00% of Python3 online submissions for How Many Numbers Are Smaller Than the Current Number.
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        cnts = [0] * 101
        res = [None] * len(nums)
        for num in nums:
            cnts[num] += 1
        for i in range(1, 101):
            cnts[i] += cnts[i - 1]
        for i in range(len(nums)):
            if nums[i] == 0:
                res[i] = 0
            else:
                res[i] = cnts[nums[i] - 1]
        return res

File: Algorithms/C++/1365-How_Many_Numbers_Are_Smaller_Than_the_Current_Number.cpp
// Runtime: 40 ms, faster than 47.66% of C++ online submissions for How Many Numbers Are Smaller Than the Current Number.
// Memory Usage: 10.4 MB, less than 100.00% of C++ online submissions for How Many Numbers Are Smaller Than the Current Number.
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        vector<int> res;
        for (int i = 0; i< nums.size(); ++i) {
            int cur = nums[i],
                smaller_num = 0;
            for (int j = 0; j < nums.size(); ++j) {
                if (cur > nums[j]) {
                    smaller_num++;
                }
            }
            res.push_back(smaller_num);
        }
        return res;
    }
};
